src/Compilers/CSharp/Portable/Binder/PatternExplainer.cs

```diff
@@ -403,6 +403,12 @@ string tryHandleListPattern(ref bool unnamedEnumValue)
                     var lengthTemp = new BoundDagTemp(lengthOrCount.Syntax, lengthOrCount.Property.Type, lengthOrCount);
                     var lengthValues = (IValueSet<int>)computeRemainingValues(ValueSetFactory.ForLength, getArray(constraintMap, lengthTemp));
                     int lengthValue = lengthValues.Sample.Int32Value;
+                    if (lengthValue > 20)
```

Comment:
>lengthValue > 20 [](http://example.com/codeflow?start=24&length=16)

I was expecting the maximum length for a list pattern example would be small, more like 3 elements or less. Beyond 3 elements or so, I'd be more inclined to write a pattern as `{ Length: N }` rather than `[_, _, _, ..., _]`.

Reply:
I think you're suggesting to limit the length to three elements *if* this is a length-only match.
What about when elements are also matched?

Reply:
Ok, here's the scenario: `_ = a switch { {Length:<20} => 0, [<0,..] => 0};`
Note that this will generate long patterns in the current revision as well.